lib/iterator.js

```diff
@@ -467,6 +480,26 @@ class SkipWhileIterator extends Iterator {
   }
 }
 
+class ChunksIterator extends Iterator {
+  constructor(base, fn) {
+    super(base);
+    this.fn = fn;
+    this.value = [];
+  }
+
+  next() {
+    let next = this.base.next();
+    if (next.done) return next;
+    while (!next.done && (!this.fn(next.value) || this.value.length === 0)) {
```

Comment:
Extract `this.value.length === 0` to `firstIteration`?

Reply:
I can't do that because it will lead to this cycle always ignoring `this.fn` calls on the first `next` iteration and therefore always consuming whole iterator on the first iteration.